### PR TITLE
CVSL-428 variation journey changes - use licence events for reasons for variation and referral

### PR DESCRIPTION
**Pull request by timharrison-moj**
This PR:
* Implements the conversation between COM and ACO during the approval or referral of variations
* Uses the new API /events/match endpoint to retrieve the conversation items
* Updates the variation flow to include the conversation in variation summaries - variation and approval
* Changes the display of a variation after approval - to highlight the conversation that took place.

NOTE: No integration tests for variation approval and referral as yet. I think there will be design changes to take on board so will do these based on feedback and then implement two more ITs.

### CODE REVIEW

File: server/routes/varyingLicences/handlers/viewVariation.ts [hunk 1]
```diff
@@ -21,6 +23,28 @@ export default class ViewVariationRoutes {
...
+        conditionComparison,
+        callToActions: { shouldShowVaryButton, shouldShowPrintButton, shouldShowEditAndDiscardButton },
+      })
+    }
+
+    // Viewing an ACTIVE licence - show conditions and prompt to vary
+    const expandedLicenceConditions = expandAdditionalConditions(licence.additionalLicenceConditions)
+    const expandedPssConditions = expandAdditionalConditions(licence.additionalPssConditions)
+
     return res.render('pages/vary/viewVariation', {
```
Comment (timharrison-moj) [inline server/routes/varyingLicences/handlers/viewVariation.ts:48]: This alters which form is displayed first depending on the status of the variation - so the conversation is seen up-front for any submitted, referred or approved variation.

File: server/views/pages/create/checkAnswers.njk [hunk 2]
```diff
@@ -44,7 +43,7 @@
     <div class="govuk-grid-row">
         <div class="govuk-grid-column-full">
             {% if licence.isVariation %}
-                {{ variationDetails(licence, isInProgress) }}
+                {{ variationSummary(licence, isInProgress) }}
             {% else %}
```
Comment (timharrison-moj) [inline server/views/pages/create/checkAnswers.njk:47]: There were two macros called `variationDetails` so renamed one to be `variationSummary` to differentiate them.

File: server/views/pages/vary-approve/request-changes.njk [hunk 3]
```diff
@@ -40,7 +43,7 @@
                         classes: 'govuk-visually-hidden'
                     },
                     errorMessage: validationErrors | findError('reasonForReferral'),
-                    value: licence.reasonForReferral | fillFormResponse(formResponses.reasonForReferral)
+                    value: formResponses.reasonForReferral
                 }) }}
```
Comment (timharrison-moj) [inline server/views/pages/vary-approve/request-changes.njk:47]: No longer pre-populating a reason - each reason is a new one - unless the result of a validation error.

File: server/views/pages/vary/viewSubmitted.njk [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+    <form method="POST">
+        <input type="hidden" name="_csrf" value="{{ csrfToken }}">
+        <div class="govuk-grid-row">
+            <div class="govuk-grid-column-full">
+                {{ cta(callToActions, licence) }}
+                {{ variationDetails(licence, conversation) }}
+                {{ variedConditions(conditionComparison) }}
+                {{ cta(callToActions, licence) }}
+            </div>
```
Comment (timharrison-moj) [inline server/views/pages/vary/viewSubmitted.njk:28]: Re-using the cta() macro - hence pulled this into its own partials module.

File: server/views/pages/vary/viewVariation.njk [hunk 5]
```diff
@@ -5,46 +5,12 @@
 {% from "partials/additionalLicenceConditionsSummary.njk" import additionalLicenceConditions %}
 {% from "partials/additionalPssConditionsSummary.njk" import additionalPssConditions %}
 {% from "partials/bespokeConditionsSummary.njk" import bespokeConditions %}
+{% from "partials/varyCallToActions.njk" import cta %}
 
 {% set pageTitle = applicationName + " - Vary a licence - View licence variation" %}
 {% set pageId = "view-variation-page" %}
 {% set backLinkHref = "/licence/vary/caseload" %}
```
Comment (timharrison-moj) [inline server/views/pages/vary/viewVariation.njk:13]: The cta() macro is now used in multiple places so pulled into its own module

File: server/views/partials/variationDetails.njk [hunk 6]
```diff
@@ -1,48 +1,36 @@
 {% from "customComponents/summaryCard/macro.njk" import summaryCard %}
```
Comment (timharrison-moj) [inline server/views/partials/variationDetails.njk:2]: Updated this macro to display the conversation as well as SPO / VLO answers. May need to split these into two distinct summary cards to match the design more closely.

File: server/services/licenceService.ts [hunk 7]
```diff
@@ -426,6 +428,32 @@ export default class LicenceService {
...
+      'DESC',
+      user
+    )
+
+    const conversation = licenceEvents?.map(event => {
+      return {
+        who: `${event.forenames} ${event.surname}`,
+        when: event.eventTime,
+        comment: event.eventDescription,
+        role: event.eventType === LicenceEventType.VARIATION_SUBMITTED_REASON ? 'COM' : 'APPROVER',
```
Comment (smcveigh941) [inline server/services/licenceService.ts:450]: Some inference here but I think it works, only a COM can submit. Wondering if there are other events we will show on the conversation which the com does and which isn't submitting a variation

**Review by smcveigh941 — APPROVED** [review]
LGTM

**Pull request merged**